app/_src/networking/meshexternalservice.md

```diff
@@ -0,0 +1,352 @@
+---
+title: MeshExternalService
+---
+
+{% warning %}
+This resource is experimental!
+{% endwarning %}
+
+This resource allows services running inside the mesh to consume services that are not part of the mesh.
+The `MeshExternalService` resource allows you to declare external resources instead of relying on [MeshPassthrough](/docs/{{ page.version }}/policies/meshpassthrough) or [passthrough mode](/docs/{{ page.version }}/networking/non-mesh-traffic#outgoing).
+
+{% warning %}
+Currently `MeshExternalService` resource only supports targeting by [MeshTrafficPermission](/docs/{{ page.version }}/policies/meshtrafficpermission) with [Zone Egress](/docs/{{ page.version }}/production/cp-deployment/zoneegress).
+This limitation will be lifted in the next release.
+{% endwarning %}
+
+## Configuration
+
+### Match
+
+This section specifies the rules for matching traffic that will be routed to external resources defined in `endpoints` section.
+The only `type` supported is [HostnameGenerator](/docs/{{ page.version }}/policies/hostnamegenerator) and it means that it will match traffic directed to a hostname created by the hostname generator.
```

Comment:
If there's only one supported type should the type be optional?

Reply:
yes, let's make it optional

Reply:
https://github.com/kumahq/kuma/pull/10570/files

Reply:
done